### PR TITLE
feat: [CDS-73647]: add status field and default selection in compiled mode

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Added Status Field and default selection in the compiled Mode (The labels are under design by the UX team will fix them later)

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/85c378bb-0ebb-4573-ad36-a63d3b4cebbf

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by bhavya-sinha — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx [hunk 1]
```diff
@@ -50,27 +50,38 @@ export function VariablesCompiledModeHeader(props: VariablesCompiledModeHeaderPr
...
       return {
-        label: execId.planExecutionId as string,
+        label: `${execId.runSequence} : ${execId.status}`,
         value: execId.planExecutionId as string
       }
     }),
     []
   )
 
+  executionIdSelectOptions.push({ label: 'No Execution', value: '' })
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx:60]: This label should be added to yaml.

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ interface VariablesCompiledModeHeaderProps {
 
 export function VariablesCompiledModeHeader(props: VariablesCompiledModeHeaderProps): JSX.Element {
   const { handleCompiledModeChange, isCompiledMode } = props
-  const [executionId, setExecutionId] = useState<string>()
+  const [executionId, setExecutionId] = useState<SelectOption>()
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx:28]: As the type of `executionId` has been changed from `string` to `SelectOption`, we can update the naming in better way. Eg: `executionIdOption`

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx]: Done

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx [hunk 3]
```diff
@@ -93,7 +104,7 @@ export function VariablesCompiledModeHeader(props: VariablesCompiledModeHeaderPr
           onChange={handleExecutionIdChange}
           name="executionId"
           placeholder="ExecutionID"
-          value={{ label: executionId as string, value: executionId as string }}
+          value={{ label: executionId?.label as string, value: executionId?.value as string }}
```
[the remark this answers is not in the document]
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx]: Done

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx]: Done

**Review by pankajsingh-harness — DISMISSED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Pull request merged**